test/opencv/test_reader_batching.py

```diff
@@ -0,0 +1,85 @@
+"""Test Base Reader's behavior and correctness with batching.
+"""
+import sys
+
+sys.path.insert(0, './')
+
+import pytest
+import cv2
+import numpy as np
+from vidsz.opencv import Reader
+
+VIDEO_PATHS = ['static/countdown.mp4']
+
+
+@pytest.mark.parametrize("vpath", VIDEO_PATHS)
+def test_correctness(vpath: str) -> None:
+    """Test Reader correctness by comparing with OpenCV Reader
+    (static-batching)
+
+    Args:
+        vpath (str): path to video
+    """
+    # open reader
+    reader = Reader(vpath, batch_size=32)
+
+    # opencv cv reader
+    cv2_reader = cv2.VideoCapture(vpath)
+
+    width = int(cv2_reader.get(cv2.CAP_PROP_FRAME_WIDTH))
+    height = int(cv2_reader.get(cv2.CAP_PROP_FRAME_HEIGHT))
+    expected_shape = (32, height, width, 3)
```

Comment:
Reason for static 32 batch size?

Reply:
it's for testing, we can use any size of batch here.